Scrum 22 create bar selection screen

RafaelRangelDeLaTejera opened the pull request:
## Description
Implemented the front end for the bar selection screen to the flutter mobile app and tested it for both android and apple



## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes

## Screenshots 
<img width="341" alt="image" src="https://github.com/user-attachments/assets/3f6ea0c1-25fe-4989-adde-d226ba7d3ada" />

RafaelRangelDeLaTejera committed:
added home screen with app bar

mstensby committed:
Work in Progress: Adjusted the models.py to allow for a customer to have a user ID, created at, and updated at fields to be able to track customer accounts.

mstensby committed:
SCRUM-50 Work in Progress: Creating the model and setting up the views for what the customer entails and what data should be associated with the customer

mstensby committed:
SCRUM-50 Work in Progress: Added a customer views python file that allows seperation from the views folder.

mstensby committed:
Work in Progress: Added a migration, but unable to add it to our complete DB.

RafaelRangelDeLaTejera committed:
Merge branch 'SCRUM-50-Django-Database-Customer' into SCRUM-22-create-bar-selection-screen

RafaelRangelDeLaTejera committed:
merged in the story with the database for the customer setup

RafaelRangelDeLaTejera committed:
added the testing framework and set up testing for the application both unit testing and widget testing

RafaelRangelDeLaTejera committed:
added the bar cards scrolling functionality

RafaelRangelDeLaTejera committed:
Finished story but still need to reformat the widget because I cant test the functions since the state is private and the functions are inside the state

RafaelRangelDeLaTejera committed:
added testing coverage report bash script

RafaelRangelDeLaTejera committed:
uncommited the coverage folder from git

RafaelRangelDeLaTejera committed:
Merge branch 'main' into SCRUM-22-create-bar-selection-screen

ShaunRichter reviewed (commented):
I see this PR is about the create bar selection screen, I am just curious what the point of the Customer Model and get_customers method is?

RafaelRangelDeLaTejera commented:
> I see this PR is about the create bar selection screen, I am just curious what the point of the Customer Model and get_customers method is?

Oh I forgot to delete that, that is something I was doing for my other story but I forgot to change branches. I will go in and change that. Thank you for catching that

RafaelRangelDeLaTejera committed:
removed files that I edited from the backend

RafaelRangelDeLaTejera committed:
added the migrations that I deleted from my branch

RafaelRangelDeLaTejera committed:
Matched backend to the main backend

RafaelRangelDeLaTejera commented:
> I see this PR is about the create bar selection screen, I am just curious what the point of the Customer Model and get_customers method is?

Oh wait nvm that was what I merged in from Matts story so that I could work on the login page endpoints, I will just remove that fie

> I see this PR is about the create bar selection screen, I am just curious what the point of the Customer Model and get_customers method is?

Okay I fixed this I just removed any changes I had made to the backend and just matched it to the one in main

ShaunRichter reviewed (approved):
LGTM
